File: leetcode/problems/problem-specific/217.contains-duplicate/217.contains-duplicate.go
package main

func containsDuplicate(nums []int) bool {
	if len(nums) <= 1 {
		return false
	}

	mappings := map[int]int{}
	for _, v := range nums {
		mappings[v]++
		if mappings[v] > 1 {
			return true
		}
	}

	return false
}

File: leetcode/problems/problem-specific/217.contains-duplicate/217.contains-duplicate.py
#
# @lc app=leetcode id=217 lang=python3
#
# [217] Contains Duplicate
#

# @lc code=start

# * Hash Table Solution | O(n) Time | O(n) Space
# * n -> The length of nums array

class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        nums_set = set()
        for num in nums:
            if num in nums_set:
                return True

            nums_set.add(num)

        return False

# @lc code=end
